Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -557,6 +558,37 @@ public class PBXProjGenerator {
         return variantGroup
     }
 
+    func getBundle(atPath path: Path) -> SourceFile {
+        // It is possible for several targets to depend on the same bundle.
+        // Do not attempt to create a bundle for each target.
+        if let cachedBundle = bundleFiles[path] {
+            return cachedBundle
+        }
+        let fileReference = PBXFileReference(reference: generateUUID(PBXFileReference.self, path.lastComponent),
+                                             sourceTree: .group,
+                                             explicitFileType: "wrapper.plug-in",
+                                             path: path.lastComponent, includeInIndex: 1)
+        fileReferencesByPath[path] = fileReference.reference
+        addObject(fileReference)
+        addObject(getGroup(path: path.parent(),mergingChildren: [fileReference.reference], depth: 0))
+        let buildFile =  PBXBuildFile(reference: generateUUID(PBXBuildFile.self, fileReference.reference), fileRef: fileReference.reference)
+        let source = SourceFile(path: path, fileReference: fileReference.reference, buildFile: buildFile)
+        bundleFiles[path] = source
+        return source
+    }
+
+    /// Return the top level group for a given path
```

Comment:
@yonaskolb I updated this to consider the `createIntermediateGroups` - what do you think?